simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -323,9 +341,38 @@ void EntityBase::requestSpeedChangeWithConstantAcceleration(
 void EntityBase::requestSpeedChangeWithTimeConstraint(
   const double target_speed, const speed_change::Transition transition, double acceleration_time)
 {
-  requestSpeedChangeWithConstantAcceleration(
-    target_speed, transition,
-    (target_speed - getStatus().action_status.twist.linear.x) / acceleration_time, false);
+  switch (transition) {
+    case speed_change::Transition::LINEAR: {
+      requestSpeedChange(target_speed, false);
+      requestSpeedChangeWithConstantAcceleration(
+        target_speed, transition, target_speed - getCurrentTwist().linear.x / acceleration_time,
+        false);
+      break;
+    }
+    case speed_change::Transition::AUTO: {
+      setDynamicConstraints(speed_planner_->planConstraintsFromJerkAndTimeConstraint(
+        target_speed, getCurrentTwist(), getCurrentAccel(), acceleration_time,
+        getDynamicConstraints()));
+      job_list_.append(
+        /**
+           * @brief Checking if the entity reaches target speed.
+           */
```

Comment:
Please fix the indentation of the comment.

Reply:
Fixed